### ISSUE TITLE
0.47.2 Changing struct to enum, which cannot be constructed because it has no accessible initializers

### ISSUE THREAD

**Issue by brandtdaniels** [nicklockwood/SwiftFormat#788]
0.47.2 Changing struct to enum, which cannot be constructed because it has no accessible initializers

'AppConfiguration' cannot be constructed because it has no accessible initializers

```
enum AppConfiguration {

  static func getCurrentBuildEnvironment() -> String {
    guard let currentEnvironment = AppConfiguration().getStringValues(forKey: .environmentName) else {
      fatalError("Unable to get current environment name from application configuration to URL object.")
    }

    return currentEnvironment
  }

  static func getBaseURL(for endpointType: NetworkEndpointType) -> URL {
    let endpoints = AppConfiguration().getDictionaryValues(forKey: .endpoints)

    guard let baseUrlString = endpoints?[endpointType.rawValue] else {
      fatalError("Unable to get \(endpointType.rawValue) Url from application configuration.")
    }

    guard let url = URL(string: baseUrlString) else {
      fatalError("Unable to convert \(endpointType.rawValue) Url "
        + "from application configuration to URL object.")
    }

    return url
  }

  static func baseUrlComponents(for endpointType: NetworkEndpointType) -> URLComponents {

    let endpoints = AppConfiguration().getDictionaryValues(forKey: .endpoints)

    guard let baseUrlString = endpoints?[endpointType.rawValue] else {
      fatalError("Unable to get \(endpointType.rawValue) Url from application configuration.")
    }

    guard let urlComponents = URLComponents(string: baseUrlString) else {
      fatalError("Unable to convert \(endpointType.rawValue) Url "
        + "from application configuration to URLComponents object.")
    }

    return urlComponents

  }

  static func getAuthConfigurationValues() -> [String: String] {
    guard let authParameters = AppConfiguration().getDictionaryValues(forKey: .auth) else {
      fatalError("Unable to generate auth configuration from app settings.")
    }

    return authParameters
  }
}
```

**Comment by nicklockwood** [comment]
Hi @brandtdaniels. The idea of this feature is that there is no reason to create instances of types that have no non-static members.

Your code example is doing something that I don't understand. You are calling `AppConfiguration().getStringValues(forKey: .environmentName)`, but the `getStringValues()` method  doesn't seem to exist. Is it defined in an extension somewhere else? 

In any case, you  can disable the rule using `--disable enumNamespaces` in the swiftformat configuration, or by placing this line at the top of your AppConfiguration file:

```swift
// swiftformat:disable enumNamespaces
```

**Comment by nicklockwood** [comment]
@brandtdaniels fixed in 0.47.3
